crates/extensions/tedge-p11-server/src/pkcs11.rs

```diff
@@ -194,25 +211,32 @@ impl Cryptoki {
     }
 }
 
-#[derive(Debug)]
-pub enum Pkcs11SigningKey {
-    Rsa(RSASigningKey),
-    Ecdsa(ECSigningKey),
+#[derive(Debug, Clone)]
+pub struct Pkcs11SigningKey {
+    session: PKCS11,
+    handle: ObjectHandle,
+    sigscheme: SigScheme,
 }
 
 impl SigningKey for Pkcs11SigningKey {
     fn choose_scheme(&self, offered: &[SignatureScheme]) -> Option<Box<dyn Signer>> {
-        match self {
-            Self::Ecdsa(e) => e.choose_scheme(offered),
-            Self::Rsa(r) => r.choose_scheme(offered),
+        debug!("Offered signature schemes. offered={:?}", offered);
+        // safe: we only create RSA and ECDSA SigningKeys
```

Comment:
I don't understand what this is referring to. None of the code indicates to me it's potentially going to do anything unsafe/panic.

Reply:
Oh, my bad, missed that one. Previously there was an unwrap somewhere here because I converted between `rustls::SignatureScheme` and `cryptoki::MechanismType` and these functions were partial, but later on I added `SigScheme` which removed those unwraps. Going to remove that comment.